src/ParallelDriver.php

```diff
@@ -44,6 +69,49 @@ public function open(string $path, string $mode): Promise
         });
     }
 
+    private function selectWorker(): \Generator
+    {
+        yield $this->pendingWorker; // Wait for any currently pending request for a worker.
+
+        if ($this->workerStorage->count() < $this->workerLimit) {
+            $worker = $this->pool->getWorker();
+            if ($worker instanceof Promise) { // amphp/parallel v1.x returns a Worker instead of a Promise.
+                $this->pendingWorker = $worker;
+                $worker = yield $worker;
+            }
+
+            if ($this->workerStorage->contains($worker)) {
+                // amphp/parallel v1.x may return an already used worker from the pool.
+                $this->workerStorage[$worker] += 1;
+            } else {
+                // amphp/parallel v2.x should always return an unused worker.
+                $this->workerStorage->attach($worker, 1);
```

Comment:
Why should it always return a new worker?

Reply:
Not new, but unused. If the worker was being held by the file driver, it won't be returned from the pool because it's busy.